### PR TITLE
get true last id from chain

### LINKED ISSUE #8535: Artist unable to mint
The artist is receiving this error while minting NFT on AHK.
Address: GyrKsk1toYMPD7dWwgTZcJf1TuBVY59UZPeU7xGCf89QMEu 

The artist tried again later, and minting worked. Still, this notification was unclear for the artist.

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

kudos to @vikiival for guidance
https://github.com/kodadot/nft-gallery/issues/8535#issuecomment-1856315938

## PR Type

- [x] Bugfix

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #8535

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 48512a3</samp>

Fixed a bug in `mintToken` composable that caused duplicate token ids by using a more reliable way of getting the last index used in a collection. Updated `lastIndexUsed` and `getNextTokenIdOnChain` functions in `composables/transaction/mintToken/utils.ts`.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 48512a3</samp>

> _`lastIndexUsed` changed_
> _Querying all item ids_
> _Fixes minting bug_

### CODE REVIEW

**Review by roiLeo — APPROVED** [review]

File: composables/transaction/mintToken/utils.ts [hunk 1]
```diff
@@ -101,11 +101,12 @@ export const assignIds = <T extends TokenToMint>(
...
   return Math.max(lastIndexUsed, lastOnChainIndex)
 }
 
-const getNextTokenIdOnChain = (api, collectionId) =>
-  api.query.nfts
-    .collection(collectionId)
-    .then((res) => Number(res.toHuman().items) || 0)
+const getLastIndexUsedOnChain = async (api, collectionId) => {
+  const collectionItems = await api.query.nfts.item.entries(collectionId)
+  const itemIds = collectionItems.map(([key]) => Number(key.args[1].toHuman()))
```
Comment (vikiival) [inline composables/transaction/mintToken/utils.ts:110]: `toHuman` on token Id 1000 will produce `1,000`

```js
x = Number('1,000')
console.log(x)
assert(x === NaN)
```

**Review by vikiival — CHANGES_REQUESTED** [review]
use toNumber, toBigInt or toString

**Comment by prury** [comment]
made a mint->burn->mint->burn inside a collection

collection:
https://deploy-preview-8583--koda-canary.netlify.app/ahk/collection/276

whole process: 

https://github.com/kodadot/nft-gallery/assets/36627808/18e34e63-16a3-417d-b147-93c9d1f947a8


transactions: 

https://assethub-kusama.subscan.io/tx/0x2bc44de6bf65cbe55ecff9ef5dfd41b64de5694bf0e4ad29b400ececbf484c57
https://assethub-kusama.subscan.io/tx/0x09aa37cf266c720f227f52a1f3dc6d300ada8dc1a61802d0cc5928093db0d63d
https://assethub-kusama.subscan.io/tx/0x067039205c49f3b5a3b530c860c3a727cd7bfdd63d8aa3cbacb48edbbd1c1f14
https://assethub-kusama.subscan.io/tx/0x614f02fdedcf27e9cdf889f918cd01ee680d23a488d30d43813d24d36aa8a24a

**Comment by daiagi** [comment]
thx @vikiival 
done

**Comment by yangwao** [comment]
Thanks!
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 7.03 USD/DOT ~ 4.267 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x5a7c7020444cbcf04c22ad54e1dc7d291f3374a64196edec9c0921d74e3fea80](https://polkadot.subscan.io/extrinsic/0x5a7c7020444cbcf04c22ad54e1dc7d291f3374a64196edec9c0921d74e3fea80)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**